tests/foreman/api/test_remoteexecution.py

```diff
@@ -0,0 +1,107 @@
+"""Test for Remote Execution
+
+:Requirement: Upgraded Satellite
+
+:CaseAutomation: Automated
+
+:CaseLevel: Acceptance
+
+:CaseComponent: API
+
+:TestType: Functional
+
+:CaseImportance: High
+
+:Upstream: No
+"""
+from nailgun import client
+from nailgun import entities
+from nailgun.entity_mixins import TaskFailedError
+
+from robottelo.api.utils import wait_for_tasks
+from robottelo.config import settings
+from robottelo.decorators import tier3
+from robottelo.decorators import tier4
+from robottelo.helpers import add_remote_execution_ssh_key
+from robottelo.vm_capsule import CapsuleVirtualMachine
+
+
+@tier4
+def test_positive_run_capsule_upgrade_playbook():
+    """Run Capsule Upgrade playbook against an External Capsule
+
+    :id: 9ec6903d-2bb7-46a5-8002-afc74f06d83b
+
+    :steps:
+        1. Create a Capsule VM, add REX key.
+        2. Run the Capsule Upgrade Playbook.
+
+    :expectedresults: Capsule is upgraded successfully
+
+    :CaseImportance: Medium
+    """
+    with CapsuleVirtualMachine() as capsule_vm:
```

Comment:
So what is the initial version of the capsule? Isn't this an upgrade from latest to latest?

Reply:
Yes it is, it runs 6.8.z upgrade, from latest to latest. Sorry, I should have phrased it better in the PR caption.

Reply:
I think what @pondrejk meant is that you don't set Capsule version explicitly when creating VM, so it might be the same version that is 6.8.z latest. So when you run `satellite-maintain`, it tries to upgrade from 6.8.1 to 6.8.1, and finishes early, but without error. And that doesn't seem to be what you want, you probably want to upgrade from 6.8.0 to 6.8.1.

Reply:
Actually the 6.8.1 -> 6.8.1 (update the latest 6.8.z) path is intentional here. The playbook triggers foreman-maintain with the provided parameters, f-m performs the upgrade (there is 6.8.z scenario) and playbook shows output and/or errors. Other upgrade paths are covered in upgrades, using foreman-maintain as well. The playbook is only a 'wrapper' for foreman-maintain, so it is not necessary to test other paths here.